src/util/LockUtils.ts

```diff
@@ -0,0 +1,56 @@
+import { getLoggerFor } from '../logging/LogUtil';
+import { InternalServerError } from './errors/InternalServerError';
+
+const logger = getLoggerFor('LockUtil');
+
+/**
+ * Waits a set amount of time, without consuming cpu, with a set amount of jitter.
+ * @param delay - How long to wait.
+ * @param jitter - A fraction of this jitter will be added to the delay.
+ * @returns A promise that resolves after the specified amount of time.
+ */
+export async function setJitterTimeout(delay: number, jitter = 0): Promise<void> {
+  jitter = Math.max(0, Math.floor(Math.random() * jitter));
+  delay = Math.max(0, delay + jitter);
+  return new Promise<void>((resolve): any => setTimeout(resolve, delay));
+}
+
+export interface AttemptSettings {
+  /** How many times should an operation be retried. (-1 is indefinitely). */
+  retryCount?: number;
+  /** The how long should the next retry be delayed (+ some retryJitter) (in ms). */
+  retryDelay?: number;
+  /** Add a fraction of jitter to the original delay each attempt (in ms). */
+  retryJitter?: number;
+}
+
+/**
+ * Will execute the given function until one of the following cases occurs:
+ * * The function resolves to a value: the resolved Promise is returned.
```

Comment:
```suggestion
 * * The function resolves to a value: the value is returned.
```
It will again be wrapped in a promise though since `retryFunction` is async so would not mention anything about being resolved to prevent confusion.